projects/storefrontlib/src/layout/a11y/keyboard-focus/visible/visible-focus.directive.ts

```diff
@@ -0,0 +1,35 @@
+import { Directive, HostBinding, HostListener, Input } from '@angular/core';
+import { BaseFocusDirective } from '../base/base-focus.directive';
+import { VisibleFocusConfig } from '../keyboard-focus.model';
+
+/**
+ * Directive implementation that adds a CSS class to the host elmeent
+ * when the moused is used to focus an element. As soon as the keyboard
+ * is used, the class is removed.
+ */
+@Directive({
+  selector: '[cxVisibleFocus]',
+})
+export class VisibleFocusDirective extends BaseFocusDirective {
+  protected defaultConfig: VisibleFocusConfig = { disableMouseFocus: true };
```

Comment:
It's confusing to me, why do we exactly have `defaultConfig` when we don't use it (eg. below in line 33) - thanks to this we could then drop explicitly setting `disableMouseFocus=true` in `storefront.component`

Reply:
Understand, it's a bit complex. We offer currently to use the `cxVisibleFocus` directly, but I'm leaning towards hiding it, and only offer the `cxFocus` directive. In the case of using `cxVisibleFocus`, the user would expect `disableMouseFocus: true` to be enabled by default, however in case of using the `cxFocus`, you'd expect to set it explicitly. 

btw, parsing the `defaultConfig` is done in the `baseFocusDirective`.